### PR TITLE
Cobolt Improvements.

### PR DESCRIPTION
**Pull request by ichristen**
Cobolt is spelled correctly and now connects to laser via USB through hwserver.

### CODE REVIEW

File: Modules/+Sources/Cobolt_PB.m [hunk 1]
```diff
@@ -0,0 +1,161 @@
+classdef Cobolt_PB < Modules.Source
+    % Cobolt_PB controls the Cobolt via USB and fast diode modulation via pulseblaster
+    
+    properties(SetObservable, GetObservable)
+        cobolt_ip = Prefs.String('No Server', 'set', 'set_cobolt_ip', 'help', 'IP/hostname of computer with hwserver for velocity laser');
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: let's change `IP` everywhere in this file to `host`

File: Modules/+Sources/Cobolt_PB.m [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        CW_power = Prefs.Double(35, 'set', 'set_power', 'min', 0, 'max', 35, 'unit', 'mW');
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: let's remove hardcoded max unless it is in their documentation for all cobolts

File: Modules/+Sources/Cobolt_PB.m [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        diode_on = Prefs.Boolean(false, 'set', 'set_diode_on', 'help', 'Power state of diode (on/off)');
+        
+        diode_sn = Prefs.Double(NaN, 'get', 'get_diode_sn', 'allow_nan', true, 'readonly', true, 'help', 'Serial number for the diode');
+        diode_age = Prefs.Double(NaN, 'get', 'get_diode_age', 'allow_nan', true, 'readonly', true, 'unit', 'hrs', 'help', 'Recorded on-time for the diode');
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: no `get` methods for prefs.

File: Modules/+Sources/Cobolt_PB.m [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        temperature = Prefs.Double(NaN, 'set', 'get_temperature', 'allow_nan', true, 'readonly', true, 'unit', 'C', 'help', 'Baseplate temperature');
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: typo?

File: Modules/+Sources/Cobolt_PB.m [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            end
+            obj = Object;
+        end
+    end
+    methods
+        function task = inactive(obj)
+            try % If PB or line is incorrect, just eat that error
+                obj.off;
+                obj.blackout;
+                task = 'Turned PB line off';
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: You turn the diode off too right?

File: Modules/+Sources/Cobolt_PB.m [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                obj.source_on = false;
+                return
+            end
+            err = [];
+            try
+                obj.PulseBlaster = Drivers.PulseBlaster.StaticLines.instance(val); %#ok<*MCSUP>
+                obj.source_on = obj.PulseBlaster.lines(obj.PB_line);
+                delete(obj.listeners)
+                obj.listeners = addlistener(obj.PulseBlaster, 'running', 'PostSet', @obj.isRunning);
+                obj.isRunning;
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: `isRunning` doesn't seem to set `obj.source_on`, should it?

File: Modules/+Sources/Cobolt_PB.m [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+            catch err
+                obj.PulseBlaster = [];
+                delete(obj.listeners)
+                obj.source_on = false;
+                val = 'No Server';
+            end
+            if ~isempty(err)
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: if there is no code between the catch statement and this if clause, you might as well rethrow error at end of catch clause and remove `err=[]` as well as this whole if block

File: Modules/+Sources/Cobolt_PB.m [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                rethrow(err)
+            end
+        end
+        function val = set_cobolt_host(obj,val,~) %this loads the hwserver driver
+            if strcmp('No Server', val)
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: I think it is worth deleting `obj.serial` every time this is called (it isn't singleton)

File: Modules/+Sources/Cobolt_PB.m [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+        
+        function arm(obj)
+            obj.diode_on = true;
+        end
+        function blackout(obj)
+            obj.diode_on = false;
+        end
+        
+        function delete(obj)
+            delete(obj.listeners)
```
Comment (mwalsh161) [inline Modules/+Sources/Cobolt_PB.m]: should also clean up `obj.serial`

File: Modules/+Sources/Cobolt_PB.m (hunk 6 above)
Comment (ichristen) [reply Modules/+Sources/Cobolt_PB.m]: I don't think so. Most of the things you are pointing out were inherited from Cobalt532_PB.m

File: Modules/+Sources/Cobolt_PB.m (hunk 7 above)
Comment (ichristen) [reply Modules/+Sources/Cobolt_PB.m]: It's fine. This was inherited. I don't want to touch it in fear of having it break. Also, it makes it safer to add things.

File: Modules/+Sources/Cobolt_PB.m (hunk 6 above)
Comment (mwalsh161) [reply Modules/+Sources/Cobolt_PB.m]: ah yes, line 127 has it.

**Pull request merged**